target-app/Managers/LocationManager.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  LocationManager.swift
+//  target-app
+//
+//  Created by Andres Leonel Torrico Cossio on 09/09/2022.
+//
+
+import UIKit
+import CoreLocation
+
+final class LocationManager: NSObject, CLLocationManagerDelegate {
+
+    static let shared = LocationManager()
+    private var locationManager: CLLocationManager = CLLocationManager()
+    private var requestLocationAuthorizationCallback: ((CLAuthorizationStatus) -> Void)?
+
+    public func requestLocationAuthorization() {
+        self.locationManager.delegate = self
+        let currentStatus = locationManager.authorizationStatus
+
+        // Only ask authorization if it was never asked before
+        guard currentStatus == .notDetermined else { return }
+
+        // Starting on iOS 13.4.0, to get .authorizedAlways permission, you need to
+        // first ask for WhenInUse permission, then ask for Always permission to
+        // get to a second system alert
+        if #available(iOS 13.4, *) {
```

Comment:
Small question here: the minimum target is 13 in your project? Could this be pushed up to 14 now that iOS 16 is out? That way you'll need to bother about those checks for previous versions 🤔

Reply:
Actually the minimum target is 15.5, so its true that this validation is unnecessary. Removed it!